ddi/src/ddi_client_impl.cpp

```diff
@@ -58,14 +58,19 @@ namespace ddi {
         };
     };
 
-    [[noreturn]] void HawkbitCommunicationClient::run() {
+    void HawkbitCommunicationClient::stop() {
+
+        isRunning = false;
+    }
+
+     void HawkbitCommunicationClient::run() {
         if (hawkbitURI.isEmpty()) {
             if (!authErrorHandler)  throw client_initialize_error("endpoint or AuthErrorHandler is not set");
             authErrorHandler->onAuthError(
                     std::make_unique<AuthRestoreHandler_>(this));
         }
 
-        while (true) {
+        while (isRunning) {
             ignoreSleep = false;
             doPoll();
             if (!ignoreSleep && currentSleepTime > 0)
```

Comment:
In addition it would be good to exit from wait_for when stop is requested, because currentSleepTime may be too long.
Consider using std::condition_variable::wait_for.

Reply:
It is still hanging on wait_for regardless of isRunning flag.
I meant that calling Stop method should also break the waiting. 
I think this is possible  by using std::condition_variable::wait_for() instead of std::this_thread::sleep_for(). though I'm not an expert in c++

Reply:
https://github.com/rtsoft-gmbh/up2date-cpp/pull/35/commits/1333e882d72a851bc56131a6e2501165fa7a3f8d

Reply:
Just check pereodically this flag. I guess full reactive reaction is unnecessary